centreon/tests/e2e/features/Api-Token/06-api-token-filter/index.ts

```diff
@@ -0,0 +1,218 @@
+        }
+      }).then((response) => {
+        expect(response.status).to.eq(201);
+      });
+    });
+  });
+});
+
+Given('I am on the API tokens page', () => {
+  cy.navigateTo({
+    page: 'API Tokens',
+    rootItemNumber: 4
+  });
+  cy.wait('@getTokens');
+
+  cy.getByLabel({ label: 'Refresh', tag: 'button' }).click();
+  cy.wait('@getTokens');
+});
+
+When('I filter tokens by {string} and click on Search', (filterBy: string) => {
+  if (filterBy == 'Name') {
+    cy.getByTestId({ testId: 'inputSearch', tag: 'input' }).type(
+      tokensToSearch.name
+    );
+    cy.getByTestId({ testId: 'inputSearch', tag: 'input' }).trigger('keydown', {
+      keyCode: 13,
+      which: 13
+    });
+    return;
+  }
+
+  cy.getByLabel({ label: 'Filter options', tag: 'button' }).click();
+
+  if (filterBy == 'Status') {
+    tokensToSearch.status == 'Active'
+      ? cy.contains('Active tokens').click()
+      : cy.contains('Disabled tokens').click();
+  } else {
+    cy.getByLabel({ label: filterBy, tag: 'input' }).click();
+    switch (filterBy) {
+      case 'Creator':
+        cy.wait('@getTokens');
+        cy.contains('li', tokensToSearch.creator).click();
+        cy.getByLabel({ label: filterBy, tag: 'input' }).click();
+        break;
+      case 'User':
+        cy.wait('@getUsers');
+        cy.contains('li', tokensToSearch.user).click();
+        cy.getByLabel({ label: filterBy, tag: 'input' }).click();
+        break;
+      case 'Creation date':
+        cy.contains('li', tokensToSearch.creationDate).click();
+        break;
+      case 'Expiration date':
+        cy.contains('li', tokensToSearch.expirationDate).click();
+        break;
+    }
+  }
+
+  cy.getByTestId({ testId: 'Search', tag: 'button' }).click();
+  cy.getByLabel({ label: 'Filter options', tag: 'button' }).click();
+});
+
+Then(
+  'I should see all tokens with a {string} according to the filter',
+  (filterBy: string) => {
+    cy.wait('@getTokens');
+
+    cy.waitUntil(
+      () => {
+        let allPromisesResolved: boolean[] = [];
+        return cy
+          .get('.MuiTableBody-root .MuiTableRow-root')
+          .each(($row) => {
+            cy.wrap($row)
+              .find('.MuiTableCell-body')
+              .eq(columnsFromLabels.indexOf(filterBy))
+              .invoke('text')
+              .then((value) => {
+                if (filterBy.includes('date')) {
+                  const [prefix] = filterBy.split(' ');
+                  allPromisesResolved.push(
+                    prefix == 'Creation'
+                      ? new Date(value.trim()) >=
+                          getDateBasedOnFilter(tokensToSearch.creationDate)
+                      : new Date(value.trim()) <=
+                          getDateBasedOnFilter(tokensToSearch.creationDate)
+                  );
+                } else {
+                  switch (filterBy) {
+                    case 'Status':
+                      allPromisesResolved.push(value == tokensToSearch.status);
```

Comment:
please always use `===` instead of `==`
I guess eslint should alert this
```suggestion
                      allPromisesResolved.push(value === tokensToSearch.status);
```